core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
Accidentally commented on the wrong PR, here's the snippet:

In regards to NotThreadSafe, Channel does use @ ThreadSafe, so there is at least some precedent. Additionally, since there is synchronization going on in AbstractStream, a new comer might be tempted to think that Streams are threadsafe.